pkg/configs/db/postgres/postgres.go

```diff
@@ -74,12 +75,18 @@ func New(uri, migrationsDir string) (DB, error) {
 	}
 
 	if migrationsDir != "" {
+		m, err := migrate.New(migrationsDir, uri)
+		if err != nil {
+			return DB{}, errors.Wrap(err, "database migrations initialization failed")
+		}
+
 		level.Info(util.Logger).Log("msg", "running database migrations...")
-		if errs, ok := migrate.UpSync(uri, migrationsDir); !ok {
-			for _, err := range errs {
-				level.Error(util.Logger).Log("err", err)
+
+		if err := m.Up(); err != nil {
+			if err != migrate.ErrNoChange {
+				return DB{}, errors.Wrap(err, "database migrations failed")
 			}
-			return DB{}, errors.New("database migrations failed")
+			level.Warn(util.Logger).Log("msg", "no change in schema, error (ignored)", "error", err)
```

Comment:
Should this be a warn level log or even a debug level log since it is expected behavior?